### ISSUE TITLE
Review comments on this repo from BSPC

### ISSUE THREAD

**Issue by XiaoPika** [MediImagePro/C2F-Net#1]
With the best of intentions, I have chosen not to recommend rejection at this stage. Instead, I would like to offer the authors an opportunity to provide clarifications, with the expectation that the manuscript can be revised to meet the quality standards of the target journal.

First, to be clear at the outset: this issue raised here are expected to be properly addressed by the authors. The authors should not attempt to delete, conceal, or alter any related data, especially given that the repository link has already been included in the submitted manuscript and is associated with immutable GitHub timestamps. In addition, I will request that this repository and the current issue be archived via web.archive.org.

The detailed comments are as follows:

Since the authors have decided to release the code publicly, they should adhere to the principles of scientific rigor and reproducibility, starting with a basic understanding of Git usage. Even a superficial inspection of the repository reveals several issues:

1. Even when uploading packaged code, it is inappropriate to include `__pycache__` directories. More concerning is the presence of a file within `__pycache__` whose class name contains `MECS`, which does not appear anywhere in the source code. This raises concerns about the integrity of the repository.

2. The repository is largely forked from https://github.com/Beckschen/TransUNet , which should be explicitly acknowledged both in the repository and in the manuscript. Furthermore, simply cloning the full TransUNet implementation and adding two Python files (`networks/CMFEM.py`, `networks/CFIFM.py`) before releasing it as the implementation of C2F-Net is likely to raise concerns among reviewers regarding the originality of the work.

3. Another critical issue, which also reflects a lack of rigor, is the presence of a hard-coded path in `train.py`:
`D:\cd\TransUNet-full\model\vit_checkpoint\imagenet21k_R50+ViT-B_16.npz.`

<br>

Beyond these structural concerns, attempts to reproduce the code revealed several issues, including but not limited to serious engineering flaws that call into question the scientific validity of the work:

1. The dataset loader remains `Synapse_dataset`, with no evidence of ultrasound-specific adaptations.

2. The CMFEM implementation includes mechanisms that are not described in the manuscript. In particular, the per-sample branch structure breaks batch-level parallelism, which directly impacts both performance and computational cost analysis. As a result, it would be impossible to reproduce the results reported in Table 2 if one strictly follows Equations (4) and (5) in the paper.
```
self.group_predictor = nn.Sequential(
    nn.AdaptiveAvgPool2d(1),
    nn.Conv2d(in_channels, in_channels // reduction_ratio, 1),
    nn.GELU(),
    nn.Conv2d(in_channels // reduction_ratio, 1, 1),
    nn.Sigmoid()
)

# 为不同分组数准备的GroupNorm和Conv层
self.group_norms = nn.ModuleDict()
self.group_convs = nn.ModuleDict()
for num_groups in range(min_groups, max_groups + 1, 4):
    self.group_norms[str(num_groups)] = nn.GroupNorm(num_groups, in_channels)
    self.group_convs[str(num_groups)] = nn.Conv2d(in_channels, in_channels, 1, groups=num_groups)
```

3. In the code, BACB is implemented as `attention × x`, whereas Equation (3) in the manuscript does not include the multiplicative factor `x`. This again suggests a lack of consistency and rigor.
```
output = self.fusion(torch.cat([att * x, group_feat], dim=1))
```

4. A closer examination of the logic in `CFIFM.py` strongly suggests that this module may have been adapted from an existing remote sensing change detection method (e.g., TFAM or similar architectures), with only superficial renaming. What is the original source of this module? What is the theoretical justification for directly transferring a bi-temporal fusion attention mechanism to encoder–decoder skip connections? Compared to the original module, what meaningful modifications have been made? <br><br>Regardless of whether the authors drew inspiration from prior work, reused code, or believe that cross-domain adaptation is justified, these aspects must be clearly explained and discussed in the manuscript.

5. Based on a preliminary reading of the manuscript, `Dataset A` appears to be critical to supporting the main conclusions. However, the `dataset_config` dictionary only includes `DDTI`, with no configuration for `Dataset A`. Does this mean that the code used to generate the `Dataset A` results reported in Tables 2 and 3 has not been released?

6. One of the training parameters in `train.py` is `--seed 1234`. Does this indicate that experiments were conducted using only a single random seed (i.e., a single run)? I will raise related concerns in the recommendation section.

7. There are many additional issues. However, due to the fact that the released code is fundamentally non-reproducible, I have decided not to pursue further in-depth analysis.
